### PR TITLE
fix🐛: hmr restore error

### PR DESCRIPTION
**Pull request by MZ-Dlovely**
## Description

stop running `createPersistedState` for hot store

## Linked Issues

#79 

## Additional context

- for example, there is a store file with a store named `key` is enabled hmr. when it is updating, `pinia` will create a temporary store named `__hot:key`. the temporary store is lie in `pinia._s` but `pinia.state`, and it will be destoried after the file is updated. so, no state of it is needed to `$patch`.

- in our plugin, `createPersistedState` will be run when any store created, so i do to inscribe a check with `store.$id`. if it's start with `__hot:` as pinia gived, i will stop running `createPersistedState`.

- error in located in the first parameter of `mergeReactiveObjects` when running  `store.$patch`.it cannot get the success store.
[pinia-$patch](https://github.com/vuejs/pinia/blob/v2/packages/pinia/src/store.ts#LL301C50-L301C50)

### CODE REVIEW

File: packages/plugin/src/plugin.ts [hunk 1]
```diff
@@ -74,8 +74,17 @@ export function createPersistedState(
...
-    if (!is_original)
+    /* c8 ignore next 3 */
+    if (!is_original) {
+      const original_id = store.$id.replace('__hot:', '')
+      // @ts-expect-error `_s is a stripped @internal`
+      const original_store = pinia._s.get(original_id)
+      /* c8 ignore next 3 */
+      if (original_store)
+        // `$persist` original_store after anything finished
+        Promise.resolve().then(() => original_store.$persist())
```
Comment (prazdevs) [inline packages/plugin/src/plugin.ts:85]: if you replace $persist() with $hydrate() it seems to work 😄

**Review by prazdevs — CHANGES_REQUESTED** [review]

File: packages/plugin/src/plugin.ts (hunk 1 above)
Comment (prazdevs) [reply packages/plugin/src/plugin.ts:85]: the point of #79 was to keep persisted data after HMR 🤔  so it considers stored data the source of truth

Comment (MZ-Dlovely) [reply packages/plugin/src/plugin.ts:85]: but before store file changed, u have done hydrate. aren't u?

**Comment by hi-reeve** [comment]
any update on this?

**Comment by prazdevs** [comment]
Ok, I looked into it, gonna merge it, and clean some things, then try it out and hopefully HMR will work and we shouldnt need the HMR package anymore

**Review by prazdevs — APPROVED** [review]

**Pull request merged**